definitions/tools/mutect.cwl

```diff
@@ -22,14 +22,7 @@ requirements:
             TUMOR=`(samtools view -H $3 | grep "^@RG" | sed "s/.*SM:\([^\t]*\).*/\1/g" | head -n 1)` #Extracting the sample name from the TUMOR bam.
             NORMAL=`(samtools view -H $4 | grep "^@RG" | sed "s/.*SM:\([^\t]*\).*/\1/g" | head -n 1)` #Extracting the sample name from the NORMAL bam.
             /gatk/gatk Mutect2 --java-options "-Xmx20g" -O $1 -R $2 -I $3 -tumor $TUMOR -I $4 -normal $NORMAL -L $5 #Running Mutect2.
```

Comment:
This chunk worries me a little bit.  There could be a case where the sample name in the bam's SM field wouldn't match the sample name provided in the YAML's `readgroup` input, which looks like:

```
readgroups:
- "@RG\tID:2900998520\tPU:HNMTJDSXX.1.ACAGCAAC-AGTCAGGA\tSM:H_KA-894912-M1513147_1\tLB:H_KA-894912-M1513147_1-lg1-lib1\tPL:Illumina\tCN:WUGSC"
```

A sanity check to make sure they match (and if not, do the rename_vcf step) might be in order.  Or maybe, just rename the VCF anyway in the mutect subworkflow to ensure that it always matches.

Reply:
To clarify, I think there are three options here:
1) always run `replace_vcf_sample_name.cwl` on the vcf files from Mutect, just to make sure things match up (and don't worry about the possible mismatch with the bam header
2) Enforce that the `SM:` tags and sample name inputs match by adding a validation step 
3) Enforce that the `SM:` tags and sample name inputs match by generating or altering the readgroup string in the alignment workflow (using the sample_name provided).

Reply:
See my comment below